docs/limitations.rst

```diff
@@ -6,186 +6,250 @@ Limitations and Workarounds
-``MPI_File_sync()`` call into the "sync-barrier-sync" construct. Enable the
-``romio_synchronizing_flush`` hint to use this workaround.
+    romio_ds_read disable
+    romio_ds_write disable
 
-**Cost:** Potentially more efficient that the ``WRITE_SYNC`` and HDF5
-``FILE_SYNC`` workarounds as this will cause the application to use the
-synchronization construct required by UnifyFS everywhere the application already
-intends them to occur (i.e., whenever there is already an ``MPI_File_sync()``).
-However, if (1) any existing ``MPI_File_sync()`` calls are only meant to make
-data visible to the other processes (rather than to avoid potential conflicts)
-or (2) the application contains a mix of lone ``MPI_File_sync()`` calls along
-with the "sync-barrier-sync" construct, then this approach will result in more
-syncs than necessary.
+These hints can be set in the ``MPI_Info`` object when opening a file,
+e.g.,::
 
-----------
+    MPI_Info info;
+    MPI_Info_create(&info);
+    MPI_Info_set(info, "romio_ds_read",  "disable");
+    MPI_Info_set(info, "romio_ds_write", "disable");
+    MPI_File_open(comm, filename, amode, info, &fh);
+    MPI_Info_free(&info);
 
-.. _file-lock-label:
+or the hints may be listed in a `ROMIOHINTS ROMIO_HINTS file`, e.g.,::
 
-File Locking
-************
+    >>: cat romio_hints.txt
+    romio_ds_read disable
+    romio_ds_write disable
+
+    >>: export ROMIO_HINTS="romio_hints.txt"
 
-UnifyFS not supporting file locks results in some I/O library features to not
-work with UnifyFS.
+MPI-I/O Shared File Pointers
+""""""""""""""""""""""""""""
 
-.. topic:: Atomicity
+ROMIO uses file locking to support MPI-I/O shared file pointers.
+One cannot use MPI-I/O shared file pointers when using UnifyFS.
+Functions that use shared file pointers include::
 
-    ROMIO uses ``fcntl()`` to implement atomicity. It is recommended to disable
-    atomicity when integrating with UnifyFS. To disable, run
-    ``MPI_File_set_atomicity(fh, 0)``.
+    MPI_File_write_shared()
+    MPI_File_read_shared()
+    MPI_File_write_ordered()
+    MPI_File_read_ordered()
 
-.. topic:: Data Sieving
+---------------------------
+HDF5 Limitations
+---------------------------
 
-    It is recommended to disable data sieving when integrating with UnifyFS.
-    Even with locking support, use of data sieving will drastically increase the
-    time and space overhead within UnifyFS, significantly decreasing application
-    performance. For ROMIO, set the hints ``romio_ds_write disable`` and
-    ``romio_ds_read disable`` to disable data sieving.
+HDF5 uses MPI-I/O.
+In addition to restrictions that are specific to HDF5,
+one must follow any restrictions associated with the
+underlying MPI-I/O implementation.
+In particular, if the MPI library uses ROMIO for its MPI-I/O implementation,
+one should adhere to any limitations noted above for ROMIO.
 
-.. topic:: Shared File Pointers
+Data Consistency
+****************
```

Comment:
We should describe when HDF flushes data to UnifyFS here, which includes HDF's flush and close operations.  Others?

Reply:
Something in my head is saying there are others, but it's been a bit and I'll need to do some digging to refresh my brain.